B 8.5. Khảo sát tại 12 công ty trong một khu vực bán hàng về doanh thu (Y ), chi phí quảng
cáo (X1) và lương nhân viên tiếp thị (X2) có số liệu sau:
Y : 126 148 105 162 101 175 160 127 138 143 158 137
X1 : 17 23 18 22 14 24 23 15 16 21 22 13
X2 : 11 14 9 16 9 17 15 11 12 14 15 13
(đơn vị tính triệu đồng)
1. Giả sử mối quan hệ giữa Y với X1 và X2 có thể biểu diễn bằng hàm hồi quy tuyến tính.
Hãy ước lượng hàm này?
2. Giải thích ý nghĩa của các hệ số hồi quy về mặt lý thuyết và thống kê?
3. Tính hệ số xác định bội và hệ số xác định bội điều chỉnh và cho biết ý nghĩa?
4. Phải chăng cả hai biến X1 và X2 đều không ảnh hưởng đến Y ?
5. Để dự báo doanh thu ta nên dùng hàm nào trong các hàm sau đây:
a) yi = b0 + b1xi + ei
b) yi = b0 + b2x2 + ei
c) yi = b0 + b1xi + b2x2 + ei
6. Dự báo doanh thu trung bình của một công ty có chi phí quảng cáo là 23 triệu đồng/tháng
và lương của nhân viên tiếp thị là 15 triệu đồng/tháng với mức ý nghĩa 5%?

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [2]:
# Y là Doanh thu
Y = [126, 148, 105, 162, 101, 175, 160, 127, 138, 143, 158, 137]  
# X1 là Chi phí quảng cáo
X1 = [17, 23, 18, 22, 14, 24, 23, 15, 16, 21, 22, 13]  
# X2 là Lương nhân viên tiếp thị
X2 = [11, 14, 9, 16, 9, 17, 15, 11, 12, 14, 15, 13] 
X2

[11, 14, 9, 16, 9, 17, 15, 11, 12, 14, 15, 13]

In [3]:


dataset = pd.DataFrame.from_dict({"Y": Y, "X1": X1, "X2": X2})

1. Giả sử mối quan hệ giữa Y với X1 và X2 có thể biểu diễn bằng hàm hồi quy tuyến tính. Hãy ước lượng hàm này?

In [4]:
results = smf.ols("Y ~ X1 + X2", data=dataset).fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                      Y   R-squared:                       0.972
Model:                            OLS   Adj. R-squared:                  0.966
Method:                 Least Squares   F-statistic:                     157.9
Date:                Wed, 20 Jul 2022   Prob (F-statistic):           9.81e-08
Time:                        11:36:58   Log-Likelihood:                -32.416
No. Observations:                  12   AIC:                             70.83
Df Residuals:                       9   BIC:                             72.29
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     29.6619      6.629      4.474      0.0

c:\Users\NNguye05\AppData\Local\Programs\Python\Python310\lib\site-packages\scipy\stats\_stats_py.py:1477: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=12
  warnings.warn("kurtosistest only valid for n>=20 ... continuing "


Từ kết quả sau khi chạy chương trình ta có thể suy ra được hàm hồi quy tuyến tính giữa Y với X1, X2.
y = 29.6619 + 0.0023*x1 + 8.4842*x2

2. Giải thích ý nghĩa của các hệ số hồi quy về mặt lý thuyết và thống kê?

Từ kết quả tính toán thì có thể kết luận là Doanh thu phụ thuộc nhiều vào Lương nhân viên tiếp thị với một hệ số rất cao 8.4842 và ít phụ thuộc vào Chi phí quảng cáo với hệ số rất nhỏ(0.0023)


3. Tính hệ số xác định bội và hệ số xác định bội điều chỉnh và cho biết ý nghĩa?

Từ kết quả tính toán ta được: 

R-squared:                       0.972 : chính là hệ số xác định bội
Adj. R-squared:                  0.966 : hệ số xác định bội điều chỉnh



4. Phải chăng cả hai biến X1 và X2 đều không ảnh hưởng đến Y ?

Kết quả tính toán cho thấy cả 2 biến X1 và X2 đều có ảnh hưởng đến Y, nhưng ở mức độ khác nhau. X2 có xu hướng ảnh hưởng mạnh mẽ đến Y còn X1 thì ảnh hưởng rất hạn chế.

5. Để dự báo doanh thu ta nên dùng hàm nào trong các hàm sau đây:
a) yi = b0 + b1xi + ei
b) yi = b0 + b2x2 + ei
c) yi = b0 + b1xi + b2x2 + ei
trả lời:
 Để dự báo chính xác chúng ta cần tính toán hàm nào cho ra giá trị của Hệ số xác định bội điều chỉnh lớn nhất. Như vậy ta cần chạy thêm 2 hàm a) và b) rồi so sánh 3 hàm a) b) c) với nhau


In [5]:
# 5.a) yi = b0 + b1xi + ei hay Y ~ X1
results = smf.ols("Y ~ X1", data=dataset).fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                      Y   R-squared:                       0.583
Model:                            OLS   Adj. R-squared:                  0.541
Method:                 Least Squares   F-statistic:                     13.98
Date:                Wed, 20 Jul 2022   Prob (F-statistic):            0.00385
Time:                        11:36:58   Log-Likelihood:                -48.685
No. Observations:                  12   AIC:                             101.4
Df Residuals:                      10   BIC:                             102.3
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     56.5118     22.761      2.483      0.0

c:\Users\NNguye05\AppData\Local\Programs\Python\Python310\lib\site-packages\scipy\stats\_stats_py.py:1477: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=12
  warnings.warn("kurtosistest only valid for n>=20 ... continuing "


In [6]:
# 5.b) yi = b0 + b2x2 + ei hay Y ~ X2
results = smf.ols("Y ~ X2", data=dataset).fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                      Y   R-squared:                       0.972
Model:                            OLS   Adj. R-squared:                  0.970
Method:                 Least Squares   F-statistic:                     350.9
Date:                Wed, 20 Jul 2022   Prob (F-statistic):           4.07e-09
Time:                        11:36:58   Log-Likelihood:                -32.416
No. Observations:                  12   AIC:                             68.83
Df Residuals:                      10   BIC:                             69.80
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     29.6711      5.999      4.946      0.0

c:\Users\NNguye05\AppData\Local\Programs\Python\Python310\lib\site-packages\scipy\stats\_stats_py.py:1477: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=12
  warnings.warn("kurtosistest only valid for n>=20 ... continuing "


Tiếp câu 5 sau khi chạy mô hình:
ta thấy:
Adj. R-squared của a) là 0.541A
Adj. R-squared của b) là 0.970
và của c) là 0.966 (kết quả từ câu 1)

Như vậy kết quả ở mô hình b) yi = b0 + b2x2 + ei cho kết quả dự đoán tốt nhất. chọn phương án b.

6. Dự báo doanh thu trung bình của một công ty có chi phí quảng cáo là 23 triệu đồng/tháng và lương của nhân viên tiếp thị là 15 triệu đồng/tháng với mức ý nghĩa 5%?
Với mức ý nghĩa 5% hay 0.05 kết hợp với kết quả ở câu 1 ta được hàm y = 29.6619 + 0.0023*x1 + 8.4842*x2
=> y = 29.6619 + 8.4842*x2
với x2= 15000000
=> y = 29.6619 + 8.4842*15000000 = 156.92 (triệu đồng)


In [7]:
y = 29.6619 + 8.4842*15
y

156.92489999999998